tutorials/tutorial1.md

```diff
@@ -0,0 +1,217 @@
+
+```
+INFO:2020-01-22 20:32:05,270:AtacWorks-peak2bw] Reading input files
+INFO:2020-01-22 20:32:05,387:AtacWorks-peak2bw] Retaining 105959 of 105959 peaks in given chromosomes.
+INFO:2020-01-22 20:32:05,387:AtacWorks-peak2bw] Adding score
+INFO:2020-01-22 20:32:05,388:AtacWorks-peak2bw] Writing peaks to bedGraph file
+INFO:2020-01-22 20:32:05,855:AtacWorks-peak2bw] Writing peaks to bigWig file dsc.Mono.2400.cutsites.smoothed.200.3.narrowPeak.bw
+INFO:2020-01-22 20:32:06,042:AtacWorks-peak2bw] Done!
+```
+
+## Step 4: Read the training data and labels and save in .h5 format
+
+We take the three bigWig files containing noisy ATAC-seq signal, the clean ATAC-seq signal, and the clean ATAC-seq peak calls. For these three files, we read the values in the specified intervals, and save these values in a format that can be read by our model. First, we read values for the intervals in the training set (`hg19.50000.training_intervals.bed`), spanning all autosomes except chr10 and chr20.
+
+```
+python $atacworks/bw2h5.py \
+           --noisybw dsc.1.Mono.50.cutsites.smoothed.200.bw \
+           --cleanbw dsc.Mono.2400.cutsites.smoothed.200.bw \
+           --cleanpeakbw dsc.Mono.2400.cutsites.smoothed.200.3.narrowPeak.bw \
+           --intervals intervals/hg19.50000.training_intervals.bed \
+           --prefix Mono.50.2400.train \
+           --pad 5000 \
+           --batch_size 2000 \
+           --nonzero
+```
+This produces a .h5 file (`Mono.50.2400.train.h5`) containing the training data for the model.
+
+## Step 5: Read the validation data and labels and save in .h5 format
+
+Next we read and save the validation data for the model.
+
+```
+python $atacworks/bw2h5.py \
+           --noisybw dsc.1.Mono.50.cutsites.smoothed.200.bw \
+           --cleanbw dsc.Mono.2400.cutsites.smoothed.200.bw \
+           --cleanpeakbw dsc.Mono.2400.cutsites.smoothed.200.3.narrowPeak.bw \
+           --intervals intervals/hg19.50000.val_intervals.bed \
+           --prefix Mono.50.2400.val \
+           --pad 5000 \
+           --batch_size 2000
+```
+This produces a .h5 file (`Mono.50.2400.val.h5`) containing the validation data for the model.
+
+
+## Step 6: Train and validate a model using the parameters in the given config files
+
+We next train an AtacWorks model to learn a mapping from the noisy (50-cell) ATAC-seq signal to the clean (2400-cell) ATAC-seq signal and peak calls. The two .yaml files that we downloaded into the `configs` directory contain all the parameters that describe how to train the model. `configs/model_structure.yaml` contains parameters that control the architecture of the model and  `configs/config_params.yaml` contains parameters that control the process of training, such as the learning rate and batch size.
+
+To train the model, we supply the training and validation datasets as well as the two config files.
+
+```
+python $atacworks/main.py --train \
```

Comment:
I am getting the following error when running this command : `main.py: error: the following arguments are required: --field`. this option seems to be missing in the model_structure.yaml and is not provided as a command line option either

Reply:
this should be fixed now.

Reply:
field is added to the model_structure file. Hence it's fixed.